src/main/java/pochat/bot/Parser.java

```diff
@@ -0,0 +1,167 @@
+        boolean isChatOver = false;
+
+        if (textInput.equals("bye")) {
+            sayGoodbye();
+            isChatOver = true;
+        } else if (textInput.equals(("list"))) {
+            replyWithListOfTextsEntered();
+        } else if (textInput.startsWith("mark")) {
+            int taskIndex = Integer.parseInt(textInput.substring(5)) - 1;
+            markTaskDone(taskIndex);
+        } else if (textInput.startsWith("unmark")) {
+            int taskIndex = Integer.parseInt(textInput.substring(7)) - 1;
+            unmarkTaskDone(taskIndex);
+        } else if (textInput.startsWith("delete")) {
+            int taskIndex = Integer.parseInt(textInput.substring(7)) - 1;
+            deleteTask(taskIndex);
+        } else if (textInput.startsWith("todo")) {
+            try {
+                String taskDescription = textInput.substring(5);
+                if (taskDescription.isEmpty()) {
+                    throw new TaskDescriptionEmptyException();
+                }
+                addToListAndReply(taskDescription);
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new TaskDescriptionEmptyException();
+            }
+        } else if (textInput.startsWith("deadline")) {
+            try {
+                int byIndex = textInput.indexOf("/by");
+
+                String taskDescription = textInput.substring(9, byIndex - 1);
+                String deadline = textInput.substring(byIndex + 4);
+
+                if (taskDescription.isEmpty()) {
+                    throw new TaskDescriptionEmptyException();
+                }
+
+                addToListAndReply(taskDescription, deadline);
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new TaskDescriptionEmptyException();
+            }
+        } else if (textInput.startsWith("event")) {
+            try {
+                int fromIndex = textInput.indexOf("/from");
+                int toIndex = textInput.indexOf("/to");
+
+                String taskDescription = textInput.substring(6, fromIndex - 1);
+                String startDate = textInput.substring(fromIndex + 6, toIndex - 1);
+                String endDate = textInput.substring(toIndex + 4);
+
+                if (taskDescription.isEmpty()) {
+                    throw new TaskDescriptionEmptyException();
+                }
+
+                addToListAndReply(taskDescription, startDate, endDate);
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new TaskDescriptionEmptyException();
+            }
+        } else if (textInput.startsWith("find")) {
+            String keyword = textInput.substring(5);
+            this.taskList.printMatchingTasks(keyword);
+        } else {
+            replyToInvalidInput();
+        }
+
+        return isChatOver;
+    }
+
+    /**
+     * Prints the goodbye message to the user
+     */
+    private void sayGoodbye() {
```

Comment:
Not a violation of coding standards, but for this private method, the method was named well and is self-explanatory.